src/gui/mainwindow.cpp

```diff
@@ -4,342 +4,348 @@
+        m_errorWidget->setText(m_nvim->errorString());
+        m_errorWidget->showReconnect(m_nvim->canReconnect());
+        m_stack.setCurrentIndex(0);
+    } else if (status != 0) {
+        m_errorWidget->setText(QString("Neovim exited with status code (%1)").arg(status));
+        m_errorWidget->showReconnect(m_nvim->canReconnect());
+        m_stack.setCurrentIndex(0);
+    } else {
+        close();
+    }
 }
 void MainWindow::neovimError(NeovimConnector::NeovimError err)
 {
-	showIfDelayed();
-
-	switch(err) {
-	case NeovimConnector::FailedToStart:
-		m_errorWidget->setText("Unable to start nvim: " + m_nvim->errorString());
-		break;
-	default:
-		m_errorWidget->setText(m_nvim->errorString());
-	}
-	m_errorWidget->showReconnect(m_nvim->canReconnect());
-	m_stack.setCurrentIndex(0);
+    showIfDelayed();
+
+    switch (err) {
+        case NeovimConnector::FailedToStart:
+            m_errorWidget->setText("Unable to start nvim: " + m_nvim->errorString());
+            break;
+        default:
+            m_errorWidget->setText(m_nvim->errorString());
+    }
+    m_errorWidget->showReconnect(m_nvim->canReconnect());
+    m_stack.setCurrentIndex(0);
 }
 void MainWindow::neovimIsUnsupported()
 {
-	showIfDelayed();
-	m_errorWidget->setText(QString("Cannot connect to this Neovim, required API version 1, found [%1-%2]")
-			.arg(m_nvim->apiCompatibility())
-			.arg(m_nvim->apiLevel()));
-	m_errorWidget->showReconnect(m_nvim->canReconnect());
-	m_stack.setCurrentIndex(0);
+    showIfDelayed();
+    m_errorWidget->setText(QString("Cannot connect to this Neovim, required API version 1, found [%1-%2]")
+                           .arg(m_nvim->apiCompatibility())
+                           .arg(m_nvim->apiLevel()));
+    m_errorWidget->showReconnect(m_nvim->canReconnect());
+    m_stack.setCurrentIndex(0);
 }
 
 void MainWindow::neovimSetTitle(const QString &title)
 {
-	this->setWindowTitle(title);
+    this->setWindowTitle(title);
 }
 
 void MainWindow::neovimWidgetResized()
 {
-	// Neovim finished resizing, resize it back to the actual
-	// widget size - this avoids situations when neovim wants a size that
-	// exceeds the available widget size i.e. the GUI tells neovim its
-	// size, not the other way around.
-	if (isMaximized() || isFullScreen()) {
-		QSize size = geometry().size();
-		if (m_tree->isVisible()) {
-			size.scale(size.width() - m_tree->geometry().size().width(),
-				size.height(), Qt::IgnoreAspectRatio);
-		}
-		m_shell->resizeNeovim(size);
-	} else {
-		m_shell->resizeNeovim(m_shell->size());
-	}
+    // Neovim finished resizing, resize it back to the actual
+    // widget size - this avoids situations when neovim wants a size that
+    // exceeds the available widget size i.e. the GUI tells neovim its
+    // size, not the other way around.
+    if (isMaximized() || isFullScreen()) {
+        QSize size = geometry().size();
+        if (m_tabline_bar->isVisible()) {
+            size.scale(size.width(),
```

Comment:
The scale here is unnecessary right? You can just `size.setHeight(size.height() - m_tabline_bar->geometry().size().height())`

Reply:
> The scale here is unnecessary right? You can just `size.setHeight(size.height() - m_tabline_bar->geometry().size().height())`

You right. Enough to add this code to mainwindow.cpp : 

`if (m_tabline_bar->isVisible()) {
            size.setHeight(size.height() - m_tabline_bar->geometry().size().height());
 }`